app/lib/actions/katello/repository/remove_content.rb

```diff
@@ -52,6 +52,12 @@ def finalize
           end
         end
 
+        def clear_smart_proxy_history(repository)
+          repos_to_clear = ::Katello::Repository.where(:root_id => repository.root_id).pluck(:id)
+          sp_history = ::Katello::SmartProxySyncHistory.where(:repository_id => repos_to_clear)
+          sp_history.try(:delete_all)
+        end
```

Comment:
It looks like here you are clearing the history across all instances of that Root Repositroy, but i think we should just do it for that particular repository, Thoughts?